man/systemd-run.xml

```diff
@@ -207,6 +207,23 @@
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--working-directory=</option></term>
+
+        <listitem><para>Runs the service process with the specified working directory. Also see
+        <varname>WorkingDirectory=</varname> in
+        <citerefentry><refentrytitle>systemd.exec</refentrytitle><manvolnum>5</manvolnum></citerefentry>.</para>
+        </listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><option>--same-dir</option></term>
+        <term><option>-d</option></term>
+
+        <listitem><para>Similar to <option>--working-directory=</option> but uses the current working directory of the
+        caller for the service to execute.</para></listitem>
+      </varlistentry>
```

Comment:
Hmm, what would be the reason to *not* want this? `systemd-run` should never be used for daemons and such, just for commands. And for commands, this is what is everybody expects. When constructing pipelines with `systed-run --pipe`, it is natural to expect operation in cwd. So I'd vote to treat this a so-far unspecified detail (the docs say nothing about the directory) and  a bugfix, and simply make `--working-directory=.` and add a note to NEWS. I think we can get away with this, because when running in "/", people naturally use absolute paths anyway.

Reply:
I don't follow why it shouldn't be used for daemons, in my view, it is just a client to PID1's Bus API for transient units (and as such isn't very different from dropping a unit in /run/systemd/transient/) and from that standpoint, launching daemons isn't really a problem. I think it comes in very handy to quickly test unit configuration (and there have been requests to take unit definition from stdin and use that to run the unit, which just goes to show how people are actually using it).

Reply:
Hmm, I think the tool is useful for unning stuff "in service context", and I think that entails that by default we don't inherit stuff from the caller, and provide a completely clean, reset execution environment for the executed program, and I think this includes the working directory.

The way I see it is that we should emphasize reproducible behaviour over convenience here by default, and that includes the working directory. 

I mean, I see where you are coming from though, that's why I added a short option for this (`-d`), and the new `-S` option even implies it, but I think using systemd-run to run services is actually OK, and a clean service context is the main benefit this tool has going for it...